File: sic-api/Entities/Su/SuUserBusinessRole.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sic_api.Entities.Su;

[Index(nameof(UserBusinessId), nameof(BusinessRoleId), IsUnique = true)]
[Table("su_user_business_role")]
public class SuUserBusinessRole : BaseEntity
{
    [Required]
    [ForeignKey(nameof(UserBusiness))]
    [Column("user_business_id")]
    public Guid UserBusinessId { get; set; }
    public SuUserBusiness UserBusiness { get; set; } = default!;

    [Required]
    [ForeignKey(nameof(BusinessRole))]
    [Column("business_role_id")]
    public Guid BusinessRoleId { get; set; }
    public SuBusinessRole BusinessRole { get; set; } = default!;

    [Column("is_primary")]
    public bool IsPrimary { get; set; } = false;

    [Column("is_active")]
    public bool IsActive { get; set; } = false;

}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuUserBusinessRole.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "su_user_business_role",
       indexes = {
           @Index(name = "idx_user_business_role", columnList = "user_business_id, business_role_id", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuUserBusinessRole extends BaseEntity {

    @Column(name = "user_business_id", insertable = false, updatable = false)
    private UUID userBusinessId;

    @Column(name = "business_role_id", insertable = false, updatable = false)
    private UUID businessRoleId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_business_id", nullable = false)
    private SuUserBusiness userBusiness;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "business_role_id", nullable = false)
    private SuBusinessRole businessRole;

    @Column(name = "is_primary", nullable = false)
    private Boolean isPrimary = false;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;
}